packages/marshal/src/encodeToSmallcaps.js

```diff
@@ -0,0 +1,434 @@
+   *
+   * @param {Passable} passable
+   * @returns {SmallcapsEncoding} except that `encodeToSmallcaps` does not generally
+   * `harden` this result before returning. Rather, `encodeToSmallcaps` is not
+   * directly exposed.
+   * What's exposed instead is a wrapper that freezes the output before
+   * returning. If this turns out to impede static analysis for `harden` safety,
+   * we can always put the (now redundant) hardens back in. They don't hurt.
+   */
+  const encodeToSmallcapsRecur = passable => {
+    // First we handle all primitives. Some can be represented directly as
+    // JSON, and some must be encoded into smallcaps strings.
+    const passStyle = passStyleOf(passable);
+    switch (passStyle) {
+      case 'null':
+      case 'boolean': {
+        // pass through to JSON
+        return passable;
+      }
+      case 'string': {
+        if (startsSpecial(passable[0])) {
+          // Strings that start with a special char are quoted with `'`.
+          // Since `'` is itself a special character, this trivially does
+          // the Hilbert hotel. Also, since the special characters are
+          // a continuous subrange of ascii, this quoting is sort-order
+          // preserving.
+          return `'${passable}`;
+        }
+        // All other strings pass through to JSON
+        return passable;
+      }
+      case 'symbol': {
+        // At the smallcaps level, we prefix symbol names with `%`.
+        // By "symbol name", we mean according to `nameForPassableSymbol`
+        // which does some further escaping. See comment on that function.
+        assertPassableSymbol(passable);
+        const name = /** @type {string} */ (nameForPassableSymbol(passable));
+        return `%${name}`;
+      }
+      case 'undefined': {
+        return '#undefined';
+      }
+      case 'number': {
+        if (Number.isNaN(passable)) {
+          return '#NaN';
+        }
+        if (is(passable, -0)) {
+          return 0;
+        }
+        if (passable === Infinity) {
+          return '#Infinity';
+        }
+        if (passable === -Infinity) {
+          return '#-Infinity';
+        }
+        // All other numbers pass through to JSON
+        return passable;
+      }
+      case 'bigint': {
+        const str = String(passable);
+        return passable < 0n ? `${str}n` : `+${str}n`;
+      }
+      case 'copyRecord': {
+        // Currently copyRecord allows only string keys so this will
+        // work. If we allow sortable symbol keys, this will need to
+        // become more interesting.
+        const names = ownKeys(passable).sort();
+        return fromEntries(
+          names.map(name => [
+            encodeToSmallcapsRecur(name),
+            encodeToSmallcapsRecur(passable[name]),
+          ]),
+        );
+      }
+      case 'copyArray': {
+        return passable.map(encodeToSmallcapsRecur);
+      }
+      case 'tagged': {
+        return {
+          '#tagged': getTag(passable),
```

Comment:
Curious why the conjugation. `#tag` would be as readable and save 3 chars, no?

Reply:
Done